### PR TITLE
Implement read the docs

### PR DESCRIPTION
**Pull request by zramsay**
- consolidates the documentation into one place: http://tendermint.readthedocs.io/en/read-the-docs/
- we no longer need to update the website when docs are updated - RTD includes a docs building CI
- converted everything from `docs/*.md`to `docs/*.rst`and organized the `docs` directory in-line with idiomatic RTD 
- ports `/intro` from website into docs

### CODE REVIEW

**Comment by rigelrozanski** [comment]
Based on conversation with @zramsay 
Within the new `.rst` files there are links to other `.html` files. Within the readthedocs system this is fine because the new `.html` files are generated from the `.rst` files, So this is all fine and dandy it just means that if you happen to be reading the `.rst` files directly in the github repo, none of the links will work, I think this is OK so long as directly in the docs directly we include a `README.md` which states "these docs are intended to be used as a part of the readthedocs website" and then provides a link or somthing to that matter... so yeah let's include that in this PR plz

**Comment by rigelrozanski** [comment]
Also what's the deal with the failing tests? Is this still non-determinism issues?

**Comment by zramsay** [comment]
readme added. tests failing = non-determinism. (at time of writing latest commit has passed)

File: docs/rpc.rst [hunk 1]
```diff
@@ -0,0 +1,188 @@
+RPC
+===
+
+Coming soon: RPC docs powered by `slate <https://github.com/lord/slate>`__. Until then, read on.
```
Comment (zramsay) [inline docs/rpc.rst]: @melekes what does it take to host the slate docs live?

Comment (melekes) [reply docs/rpc.rst]: I hosted the slate docs on Heroku just to provide an example. But it could be any server.

**Comment by melekes** [comment]
Awesome work!

File: docs/rpc.rst (hunk 1 above)
Comment (zramsay) [reply docs/rpc.rst]: #629

**Pull request merged**